doc/source/data/api/dataset_iterator.rst

```diff
@@ -0,0 +1,14 @@
+.. _dataset-iterator-api:
+
+DatasetIterator API
+===================
+
+.. currentmodule:: ray.data
+
+.. autosummary::
+   :toctree: doc/
+
+  DatasetIterator.iter_batches
+  DatasetIterator.iter_torch_batches
+  DatasetIterator.to_tf
+  DatasetIterator.stats
```

Comment:
to get rid of warning of `/ray/doc/source/data/api/doc/ray.data.DatasetIterator.iter_batches.rst: WARNING: document isn't included in any toctree`, could we add a separate heading? (haven't tried it to verify it would fix the issue, but worth to try)

```
DatasetIterator API
===================

.. currentmodule:: ray.data

Constructor
-----------

.. autosummary::
   :toctree: doc/

   DatasetIterator

Iterating Datasets
------------------

.. autosummary::
   :toctree: doc/

  DatasetIterator.iter_batches
  DatasetIterator.iter_torch_batches
  DatasetIterator.to_tf

Getting Statistics
------------------

.. autosummary::
   :toctree: doc/

  DatasetIterator.stats
```

Reply:
Turns out problem was spacing :D